src/context/useModal.tsx

```diff
@@ -16,10 +16,13 @@ export const ModalProvider = ({ children }: { children: ReactNode }) => {
   const [content, setContent] = useState<ReactNode | null>(null);
   const { openPortal, closePortal, isOpen, Portal } = usePortal();
 
-  const showModal = useCallback((content: ReactNode) => {
-    setContent(content);
-    openPortal();
-  }, []);
+  const showModal = useCallback(
+    (content: ReactNode, event?: SyntheticEvent<HTMLElement>) => {
+      setContent(content);
+      openPortal(event);
```

Comment:
Can we set `programmaticallyOpen: true` instead of relaying the event here? Seems simpler:
```
  const { openPortal, closePortal, isOpen, Portal } = usePortal({
    programmaticallyOpen: true,
  });
```